webrtc-c/canary/src/Peer.cpp

```diff
@@ -313,6 +313,10 @@ STATUS Peer::shutdown()
         CHK_LOG_ERR(closePeerConnection(this->pPeerConnection));
     }
 
+    if (IS_VALID_SIGNALING_CLIENT_HANDLE(this->pSignalingClientHandle)) {
```

Comment:
Just verifying that in this case a single peer is the entire canary run so it's OK to delete the entire channel with this peer? I am trying to understand the lifecycle and pardon my ignorance here. Normally, the peers have shorter lifecycle than the signaling channel

Reply:
This should be OK since the peer connection will close first and notify the other peer to also terminate, there shouldn't be anymore communication and this kind of communication should be done in peer-to-peer manner. But, to guarantee this even further, I can add a check so that it's done only in the viewer, similar to what we have for signaling canary.